File: app/schemas/metrics.py
"""
Pydantic schemas for Observability, System Health & Queue Telemetry.
"""

from __future__ import annotations

import uuid
from pydantic import BaseModel


class QueueStatsResponse(BaseModel):
    """Real-time statistical breakdown and throughput for a specific queue."""

    queue_id: uuid.UUID
    name: str
    pending: int
    running: int
    completed_24h: int
    failed_24h: int
    dead: int
    throughput_per_hour: float


class SystemMetricsResponse(BaseModel):
    """Organization-wide health metrics and aggregated counters."""

    total_jobs: int
    active_workers: int
    total_queues: int
    jobs_completed_24h: int
    jobs_failed_24h: int
    overall_failure_rate_24h: float

File: frontend/src/api/metrics.ts
import { apiClient } from './client';

export interface QueueStats {
  queue_id: string;
  name: string;
  pending: number;
  running: number;
  completed_24h: number;
  failed_24h: number;
  dead: number;
  throughput_per_hour: number;
}

export interface SystemMetrics {
  total_jobs: number;
  active_workers: number;
  total_queues: number;
  jobs_completed_24h: number;
  jobs_failed_24h: number;
  overall_failure_rate_24h: number;
}

export const metricsApi = {
  getQueueStats: async (queueId: string): Promise<QueueStats> => {
    const res = await apiClient.get<QueueStats>(`/metrics/queues/${queueId}`);
    return res.data;
  },

  getSystemMetrics: async (): Promise<SystemMetrics> => {
    const res = await apiClient.get<SystemMetrics>('/metrics/system');
    return res.data;
  },
};
